src/Persistence/Templates/ControlFactory.cs

```diff
@@ -60,12 +63,12 @@ public App CreateApp(string name, ControlPropertiesCollection? properties = null
         };
     }
 
-    public Screen CreateScreen(string name, ControlPropertiesCollection? properties = null, Control[]? children = null)
+    public Screen CreateScreen(string name, ControlPropertiesCollection? properties = null, IList<Control>? children = null)
     {
         return new Screen(name, _controlTemplateStore)
         {
             Properties = properties ?? new(),
-            Children = children ?? Array.Empty<Control>()
+            Children = children
```

Comment:
please add fallback

Reply:
why we need fallback here and to what?

Reply:
`Enumerable.Empty`

Reply:
Some controls should never have children and having "null" value signifies that fact. I am not sure why we need to have empty list in these cases

Reply:
true. But from the code perspective, having the guarantee that the collection is always initialized (even when empty) saves from the headache of doing null-checks each time. It would lead to far less bugs.

Reply:
having empty collection will lead to different class of bugs when user can safely enumerate children when there should be no children at all.

Reply:
this is why we should expose this as array or readonly collection and add a specific `AddChildren` method. The collection itself would be always instantiated and the Add method would check if the control can have children or not

Reply:
I am not talking about adding children. I am talking about enumerating children when there should be no children at all. There should be an exception in this case

Reply:
sorry, but I disagree with that. We should throw when adding a child to a control that _can't_ have ones, not when asking "how many children do you have". As rule of thumb, queries should never throw, while commands instead can fail (for example when input validation goes wrong)

Reply:
Perhaps we should move `Children` up into a higher class?  Controls which may have children inherit from that, controls which don't instead inherit from the base?

Label -> Control
Gallery -> ControlWithChildren -> Control

Reply:
Might be idea for separate PR